File: frontend/quiz/src/seleniumTests/Pages/QuizPage.js
import Page from "./Page";
import {
  getElementById,
  findElementByText,
  getElementByXpath
} from "./Utility";

const incorrectAnswerXpath =
  "//h4[contains(text(),'Option')]/ancestor::div[contains(@id,'answer')]";
const correctAnswerXpath =
  "//h4[contains(text(),'Answer')]/ancestor::div[contains(@id,'answer')]";
const firstOptionId = "answer0";
const secondOptionId = "answer1";
const continueButtonId = "ContinueButton";
const counterId = "counterId";
class QuizPage extends Page {
  async answerFirstOptionAndSubmit() {
    await (await this.firstAnswerOptionBtn).click();
    await (await this.nextQuestionBtn).click();
  }
  get correctAnswerOptionBtn() {
    return getElementByXpath(this.driver, correctAnswerXpath);
  }

  get incorrectAnswerOptionBtn() {
    return getElementByXpath(this.driver, incorrectAnswerXpath);
  }
  get CounterCurrentQuestion() {
    return getElementById(this.driver, counterId);
    // .then(el =>      el.getText());
  }

  get firstAnswerOptionBtn() {
    return getElementById(this.driver, firstOptionId);
  }

  async findElementByTextClick(text) {
    findElementByText(text).then(el => el.click());
  }
  get nextQuestionBtn() {
    return getElementById(this.driver, continueButtonId);
  }

  async open(path) {
    await super.open(path);
  }
  get secondAnswerOptionBtn() {
    return getElementById(this.driver, secondOptionId);
  }
}

export default QuizPage;

File: backend/selenium/QuizPage.py
from Utility import getElementById, isVisibleById, getElementByText,  getElementByXpath
from Page import Page

incorrectAnswerXpath = "//h4[contains(text(),'Option')]/ancestor::div[contains(@id,'answer')]"
correctAnswerXpath = "//h4[contains(text(),'Answer')]/ancestor::div[contains(@id,'answer')]"
firstOptionId = "answer0"
secondOptionId = "answer1"
continueButtonId = "ContinueButton"
counterId = "counterId"


class QuizPage (Page):
    def answerFirstOptionAndSubmit(self):
        (self.firstAnswerOptionBtn()).click()
        (self.nextQuestionBtn()).click()

    def correctAnswerOptionBtn(self):
        return getElementByXpath(self.driver, correctAnswerXpath)

    def counterCurrentQuestion(self):
        return (getElementById(self.driver, counterId)).text

    def incorrectAnswerOptionBtn(self):
        return getElementByXpath(self.driver, incorrectAnswerXpath)

    def firstAnswerOptionBtn(self):
        return getElementById(self.driver, firstOptionId)

    def findElementByTextClick(self, text):
        (getElementByText(self.driver, text)).click()

    def nextQuestionBtn(self):
        return getElementById(self.driver, continueButtonId)

    def secondAnswerOptionBtn(self):
        return getElementById(self.driver, secondOptionId)
